mlbench_core/evaluation/goals.py

```diff
@@ -12,64 +12,57 @@ def _add_detailed_times(result, tracker):
     return result
 
 
-def task1_time_to_accuracy_light_goal(metric_name, value, tracker):
-    """ Accuracy over Time target for benchmark task 1: Image classification
-    (Light)
+def time_to_accuracy_goal(threshold):
+    def _time_to_accuracy_goal(metric_name, value, tracker):
+        if metric_name != "val_global_Prec@1":
+            return None
+        if value >= threshold:
+            duration = tracker.get_total_train_time()
 
-    Light target is 70% accuracy
+            result = "{0:02d}% Top 1 Validation Accuracy reached in {1:.3f} " \
+                     "seconds".format(threshold, duration)
 
-    Args:
-        metric_name(str): Name of the metric to test the value for,
-        only "val_Prec@1" is counted
-        value (float): Metric value to check
-        tracker (`obj`:mlbench_core.utils.tracker.Tracker): Tracker object
-        used for the current run
-    Return:
-        result (str) or `None` if target is not reached
-    """
-    if metric_name != "val_global_Prec@1":
-        return None
-
-    if value >= 70:
-        duration = tracker.get_total_train_time()
+            result = _add_detailed_times(result, tracker)
 
-        result = "70% Top 1 Validation Accuracy reached in {0:.3f} seconds" \
-            .format(duration)
-
-        result = _add_detailed_times(result, tracker)
+            return result
 
-        return result
+        return None
 
-    return None
+    return _time_to_accuracy_goal
 
 
-def task1_time_to_accuracy_goal(metric_name, value, tracker):
-    """ Accuracy over Time target for benchmark task 1: Image classification
+def task2_time_to_accuracy_goal():
```

Comment:
We should do a fixed goal for task1 too that wraps `time_to_accuracy_goal(..)` like you did it here.

Reply:
Done ! Should I merge ?